### PR TITLE
DW-4879 Add Aurora MySQL hive metastore support

### PR DESCRIPTION
**Pull request by draagc-iw**

### CODE REVIEW

File: terraform/modules/emr/templates/emr/configuration.mysql.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      "spark.driver.extraJavaOptions": "-XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=70 -XX:MaxHeapFreeRatio=70 -XX:+CMSClassUnloadingEnabled -XX:OnOutOfMemoryError='kill -9 %p' -Dhttp.proxyHost='${proxy_host}' -Dhttp.proxyPort='3128' -Dhttp.nonProxyHosts='${full_no_proxy}' -Dhttps.proxyHost='${proxy_host}' -Dhttps.proxyPort='3128'",
+      "spark.executor.extraJavaOptions": "-verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=70 -XX:MaxHeapFreeRatio=70 -XX:+CMSClassUnloadingEnabled -XX:OnOutOfMemoryError='kill -9 %p' -Dhttp.proxyHost='${proxy_host}' -Dhttp.proxyPort='3128' -Dhttp.nonProxyHosts='${full_no_proxy}' -Dhttps.proxyHost='${proxy_host}' -Dhttps.proxyPort='3128'"
+    }
+  },
+  {
+    "Classification": "spark-hive-site",
+    "Properties": {
+      "javax.jdo.option.ConnectionURL": "jdbc:mysql://${hive_metastore_endpoint}:3306/${hive_metastore_database_name}?createDatabaseIfNotExist=true",
+      "javax.jdo.option.ConnectionDriverName": "org.mariadb.jdbc.Driver",
+      "javax.jdo.option.ConnectionUserName": "username",
```
Comment (IanUtley) [inline terraform/modules/emr/templates/emr/configuration.mysql.json:50]: This supposed to be username=username and password=password?

Comment (draagc-iw) [reply terraform/modules/emr/templates/emr/configuration.mysql.json:50]: Nope, rebase error, fixing

Comment (IanUtley) [inline terraform/modules/emr/templates/emr/configuration.mysql.json:50]: Shouldnt this be ${hive_metastore_username}, similarly for password

File: terraform/modules/emr/local.tf [hunk 2]
```diff
@@ -12,6 +12,29 @@ locals {
   autoscaling_max_capacity        = 5
   dks_port                        = 8443
   full_proxy                      = var.internet_proxy["http_address"]
+
+  configurations_mysql_json = templatefile(format("%s/templates/emr/configuration.mysql.json", path.module), {
+    logs_bucket_path             = format("s3://%s/logs", var.log_bucket)
+    data_bucket_path             = format("s3://%s/data", aws_s3_bucket.emr.id)
+    notebook_bucket_path         = format("%s/data", aws_s3_bucket.emr.id)
```
Comment (IanUtley) [inline terraform/modules/emr/local.tf:19]: I cant see where this is referenced in mysql.json file

Comment (draagc-iw) [reply terraform/modules/emr/local.tf:19]: This is probably something we removed for the configuration.json a long time ago but forgot to remove the variable.

**Review by IanUtley — APPROVED** [review]

**Pull request merged**